integration-test/background/request-blocking.js

```diff
@@ -0,0 +1,59 @@
+const harness = require('../helpers/harness')
+const { logPageRequests } = require('../helpers/requests')
+const backgroundWait = require('../helpers/backgroundWait')
+const pageWait = require('../helpers/pageWait')
+
+const testSite = 'https://privacy-test-pages.glitch.me/privacy-protections/request-blocking/'
+
+let browser
+let bgPage
+let teardown
+
+describe('Test request blocking', () => {
+    beforeAll(async () => {
+        ({ browser, bgPage, teardown } = await harness.setup())
+        await backgroundWait.forAllConfiguration(bgPage)
+    })
+
+    afterAll(async () => {
+        await teardown()
+    })
+
+    it('Should block all the test tracking requests', async () => {
+        // Load the test page.
+        const page = await browser.newPage()
+        await pageWait.forGoto(page, testSite)
+
+        // Start logging network requests.
+        const pageRequests = []
+        logPageRequests(page, pageRequests)
+
+        // Initiate the test requests.
+        // Note: Waiting for network idle here does not work for some reason, so
+        //       wait for ten seconds instead.
+        await page.click('#start')
+        await backgroundWait.forTimeout(bgPage, 10000)
```

Comment:
With the above change, you should be able to remove this.

If you need to wait for the test to finish, you should be able to check for the `window.results` object.

Reply:
The `results` Object is not a property of `window` unfortunately (so we need the ESLint comment to turn off the undefined variable warning).

Also, I think the test page pushes results onto the `results.results` Array as they come in. Unless I'm missing something, we'd either need to hard-code the number of expected results in the integration tests, or we still need the timeout here. What do you think?

Reply:
Ah, scratch that. I just realised we can wait for `results.results.length` to match `tests.length` instead. I've got rid of the annoying ten second wait now.